development/views.py

```diff
@@ -52,18 +56,62 @@ def form_valid(self, form):
         return super().form_valid(form)
 
 
-class MatchListView(SingleTableView):
+class MatchListView(ListView):
     model = Match
-    table_class = MatchTable
     template_name = 'development/match_history.html'
 
     def get_queryset(self):
-        return Match.objects.filter(user_1=self.request.user) | Match.objects.filter(user_2=self.request.user)
+        return Match.objects.filter(
+            Q(user_1=self.request.user)
+            | Q(user_2=self.request.user)
+        ).order_by("-date_match")
 
 
 class MatchDetailView(DetailView):
-    model = Match
-    template_name = 'development/match_detail.html'
+    template_name = 'development/match_details.html'
+
+    def __init__(self, *args, **kwargs):
+        super(MatchDetailView, self).__init__(*args, **kwargs)
+        self.current_page = 1
+        self.prev_page = 1
+        self.next_page = 2
+        self.pages = {
+            1: '',
+            2: '',
+            3: '',
+        }
```

Comment:
If this a work in progress please add a `# TODO: Think about how to deal with pagination ` comment.

Reply:
done!